expression/builtin_math.go

```diff
@@ -515,6 +515,15 @@ func fixFloorAndCeilType(arg Expression) (retTp, argTp evalTp) {
 	return
 }
 
+// setFlag4FloorAndCeil sets return flag of FLOOR and CEIL
+func setFlag4FloorAndCeil(bf *baseBuiltinFunc, arg Expression) {
+	fieldTp := arg.GetType()
+	if (fieldTp.Tp == mysql.TypeLong || fieldTp.Tp == mysql.TypeNewDecimal) && mysql.HasUnsignedFlag(fieldTp.Flag) {
```

Comment:
other function also need such function IsTypeInt? I think current condition maybe more clear in this scenario.